### PR TITLE
Galaxy fixes

### PR DESCRIPTION
**Pull request by Hyask**
- Improved 404 page to include context.
  - Better typing to galaxy.
  - Fix error 500 for user not present in galaxy.
  - Don't display the link to user's galaxy page, as it's not ready yet for full public disclosure.
  - Update locales, including the Galaxy's

Also, this MR is rebased on `master` and targeted towards it (only the three last commits have not been reviewed yet), as the goal is likely to push it to production to gather representative data that will be used to correctly caliber graph weights (see the end of the discussion on #562).

### CODE REVIEW

File: galaxy/models.py [hunk 1]
```diff
@@ -229,15 +229,15 @@ def query_user_score(cls, user):
     ####################
 
     @classmethod
-    def compute_users_score(cls, user1, user2):
+    def compute_users_score(cls, user1, user2) -> int:
```
Comment (imperosol) [inline galaxy/models.py:232]: Indication de type fausse. La fonction retourne un `tuple[int, int, int, int]`

File: galaxy/views.py [hunk 2]
```diff
@@ -47,48 +47,51 @@ def get_queryset(self):
...
-                ),
-                other_star_mass=Case(
-                    When(star1=self.object.galaxy_user, then=F("star2__mass")),
-                    default=F("star1__mass"),
-                ),
-                other_star_name=Case(
-                    When(
-                        star1=self.object.galaxy_user,
-                        then=Case(
+        try:
```
Comment (imperosol) [inline galaxy/views.py:50]: Le try/catch:pass, c'est quelque chose qui est vraiment pas aisé à lire. En plus, ça n'a pas grand chose à faire là : la méthode `get_context_data()` devrait juste servir à construire les données qui sont renvoyées. La gestion de ce genre d'exception devrait se faire en amont.

Dans le cas où l'utilisateur n'est pas dans la galaxie, il serait peut-être plus judicieux de renvoyer dès le début une erreur 404 avec une explication de l'erreur.

**Review by imperosol — CHANGES_REQUESTED** [review]

File: galaxy/models.py (hunk 1 above)
Comment (Hyask) [reply galaxy/models.py]: Fixed, thanks

File: galaxy/views.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    template_name = "galaxy/user.jinja"
+    current_tab = "galaxy"
+
+    def get_object(self, *args, **kwargs):
+        object = super(GalaxyUserView, self).get_object(*args, **kwargs)
+        try:
+            print(object.galaxy_user)
+        except User.galaxy_user.RelatedObjectDoesNotExist:
+            raise Http404(_("This citizen has not yet joined the galaxy"))
+        return object
```
Comment (imperosol) [inline galaxy/views.py:52]:
```suggestion
    def get_object(self, *args, **kwargs):
        user: User = super(GalaxyUserView, self).get_object(*args, **kwargs)
        if not hasattr(user, "galaxy_user"):
            raise Http404(_("This citizen has not yet joined the galaxy"))
        return user
```

Comment (Hyask) [reply galaxy/views.py]: Oops, didn't finish the work here, sorry about that.

**Review by imperosol — APPROVED** [review]

**Comment by Hyask** [comment]
@TheoDurr PR is approved and ready to merge. Once deployed, please run the following command, and send me the output log file (`rule_galaxy.log`) as well as the `time` output that will be in the shell:  
```
time ./manage.py rule_galaxy -v 3 > rule_galaxy.log 2>&1
```

With those data, I'll be able to fine tune the various weights of the galaxy, add some representative procedural generated test data, and improve the graph output to make sure it's readable.

**Pull request merged**